### PR TITLE
[FIX] hr_expense: re-allow edition for approved expenses

### PR DESCRIPTION
**Pull request by JulienAlardot**
[FIX] hr_expense: re-allow edition for approved expenses

Partial revert of f892800

Context:
In f892800, too many editing rights were restricted,
not allowing accountants to properly edit important fields
that are related to the linked `account.move`s.
As there is no automatic synchronization between the two models,
the data may be different, but they may want to edit it by hand
and there is no reason why we shouldn't disallow it.

After this commit:
We revert the readonly state & re-allow edition of expenses & their 
sheets that are approved, only restricting the (un)linking of expenses 
from a sheet to accountants. To avoid having an expense paid twice.

### CODE REVIEW

File: addons/hr_expense/views/hr_expense_views.xml [hunk 1]
```diff
@@ -53,27 +53,27 @@
                     <field name="product_has_cost" column_invisible="True"/>
                     <field name="date" optional="show" readonly="not is_editable"/>
                     <field name="product_id" optional="hide" readonly="not is_editable" required="1"/>
-                    <field name="name" readonly="not is_editable"/>
+                    <field name="name" readonly="not is_editable and state != 'approved'"/>
```
Comment (svfu-odoo) [inline addons/hr_expense/views/hr_expense_views.xml:56]: After your change this field (and the others) are always editable (not readonly) when `state == 'approved'` (no matter the value of `is_editable`).
But `is_editable` also depends on user groups. Could this not cause an issue?

**Comment by JulienAlardot** [comment]
@svfu-odoo Hello, could this be reviewed ASAP? It is needed before the 06/08 if possible

File: addons/hr_expense/tests/test_expenses_access_rights.py [hunk 2]
```diff
@@ -1,10 +1,12 @@
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
 from odoo import Command
-from odoo.addons.hr_expense.tests.common import TestExpenseCommon
```
Comment (svfu-odoo) [inline addons/hr_expense/tests/test_expenses_access_rights.py:4]: (Optional) unnecessary diff

**Review by svfu-odoo — COMMENTED** [review]
Thanks for the update.

Please improve the commit / PR message a bit.
What went wrong in the linked commit and how is it different now?

**Review by svfu-odoo — APPROVED** [review]
LGTM :+1:
Thanks for the update
@JulienVR for final review

**Review by JulienVR — COMMENTED** [review]
Everything looks fine :+1:
Just 2 tiny questions

File: addons/hr_expense/models/hr_expense_sheet.py [hunk 3]
```diff
@@ -470,11 +470,19 @@ def create(self, vals_list):
 
     def write(self, values):
         res = super().write(values)
+
+        user_is_accountant = self.env.user.has_group('account.group_account_user')
+        edit_lines = 'expense_line_ids' in values
+        edit_states = 'state' in values or 'approval_state' in values
+        # Forbids (un)linking expenses from an approved sheet if you're not an accountant
+        if edit_lines and not user_is_accountant and set(self.mapped('state')) - {'draft', 'submit'}:
+            raise UserError(_("You do not have the rights to add or remove any expenses on an approved or paid expense report."))
```
[the remark this answers is not in the document]
Comment (JulienAlardot) [reply addons/hr_expense/models/hr_expense_sheet.py:479]: Tru dat

File: addons/hr_expense/i18n/hr_expense.pot [hunk 4]
```diff
@@ -2375,6 +2375,14 @@ msgstr ""
 msgid "You cannot set the expense total to 0 if it's linked to a report."
 msgstr ""
 
+#. module: hr_expense
+#. odoo-python
+#: code:addons/hr_expense/models/hr_expense_sheet.py:0
+msgid ""
+"You do not have the rights to add or remove any expenses on an approved or "
```
[the remark this answers is not in the document]
Comment (JulienAlardot) [reply addons/hr_expense/i18n/hr_expense.pot:2382]: According to our official englishman, both make as much sense

**Comment by JulienVR** [comment]
@robodoo r+
